### PR TITLE
DRY: Don't Repeat Yourself

### PR DESCRIPTION
**Pull request by tasleson**
While looking around in the code it's pretty apparent that there is **a lot** of copy and pasted code.  I'm not sure what others think about this, but I personally try to avoid it.  I took one file and worked on it a little bit today, to see what I could do.

This has only been compiled, likely I've introduced a number of bugs.  If people would like to reduce the source code size and feel better that when they fix a bug they don't have to grep the entire source base to ensure the bug isn't duplicated somewhere else I will work on this some more.  Otherwise we can toss this PR and keep doing what we are doing.

### CODE REVIEW

**Comment by vojtechtrefny** [comment]
Jenkins, ok to test.

**Comment by vojtechtrefny** [comment]
Hi, I think this is a great idea, especially the authentication part. But this is definitely not only problem for partitioning code -- the code for policykit authentication is in all modules, so the `check_authenticate` method should be in some "utils" file or something like that.

And you should probably do this on the _master-libblockdev_ branch -- we've changed quite a lot of code when rewriting this to libblockdev and we plan to merge it to master soon.

**Comment by tasleson** [comment]
@vojtechtrefny From leveraging a code duplication tool, it appears that **~21%** of the master branch is duplicated code.  I picked partition as it had some of the biggest pieces of copy and paste, but yes there is copy & pasted code all across the code base.  I'll look at this more and see if I can make the code changes more generic across the code base, like a common authentication check like you mentioned.  This will also be a good way for me to get more familiar with the code base too.

**Comment by vpodzime** [comment]
Jenkins, test this, please.

**Comment by vpodzime** [comment]
Jenkins, test this, please.

**Comment by tasleson** [comment]
Re-based to `master-libblockdev` instead of `master`.  A number of original changes don't make sense anymore.  The last commit in this series should probably **not** be included as the complexity of the change yields very little code savings and increases complexity.

File: src/udiskslinuxpartition.c [hunk 1]
```diff
@@ -102,6 +102,84 @@ udisks_linux_partition_new (void)
                                          NULL));
 }
```
Comment (vpodzime) [inline src/udiskslinuxpartition.c]: typo in the commit message - *autherization*

File: src/udiskslinuxpartition.c [hunk 2]
```diff
@@ -102,6 +102,84 @@ udisks_linux_partition_new (void)
...
+                                               NULL,
+                                               &error))
+    {
+      g_dbus_method_invocation_return_gerror (invocation, error);
+      g_clear_error (&error);
+      goto out;
+    }
+
+  /* Setup the returned caller uid */
+  *returned_caller_uid = caller_uid;
```
Comment (vpodzime) [inline src/udiskslinuxpartition.c]: Any reason why ``returned_caller_uid`` is not used in place of ``caller_uid``? Could then be simply called ``caller_uid``.

File: src/udiskslinuxpartition.c [hunk 3]
```diff
@@ -289,62 +289,31 @@ udisks_linux_partition_update (UDisksLinuxPartition   *partition,
...
+                                         GDBusMethodInvocation  *invocation);
+
+
+static gboolean _set_flags (GDBusMethodInvocation *invocation,
+                            UDisksBlock           *block,
+                            gchar                 *device_name,
+                            UDisksPartitionTable  *partition_table,
+                            gchar                 *partition_name,
+                            UDisksBaseJob         *job,
+                            void                  *user_data)
```
Comment (vpodzime) [inline src/udiskslinuxpartition.c]: No need to start identifiers with ``_``. And the above ``typedef`` definitions define types so the identifiers should be CamelCase.

**Comment by tasleson** [comment]
@vpodzime Removed last commit, corrected commit spelling, and re-worked variables in `check_authorization` function as suggested.

**Comment by vpodzime** [comment]
> @vpodzime Removed last commit, corrected commit spelling, and re-worked variables in check_authorization function as suggested.

Thanks!

**Pull request merged**